From f828289cd561b944496fa779acdfe4169f53ab9b Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Tue, 11 Feb 2025 11:31:10 +0800
Subject: [PATCH 13/14] xhpast: Skip -minline-all-stringops on non-x86
 architectures

Summary:
-minline-all-stringops is a x86 only option, and will not work on other architectures.
Thus remove it for other architectures.

Test Plan: Build on other architectures. I cherry-picked this to AOSC OS packaging and it has built successfully on amd64, arm64, riscv64, ppc64el, loongson3, loongarch64.

Reviewers: O1 Blessed Committers!

Subscribers: tobiaswiese, valerio.bozzolan, Matthew, Cigaryno

Differential Revision: https://we.phorge.it/D25871
---
 support/xhpast/Makefile                           |  2 +-
 .../bin/xhpast-generate-release-cppflags.php      | 15 +++++++++++++++
 2 files changed, 16 insertions(+), 1 deletion(-)
 create mode 100755 support/xhpast/bin/xhpast-generate-release-cppflags.php

diff --git a/support/xhpast/Makefile b/support/xhpast/Makefile
index 3686de3e2e5e..64a211745909 100644
--- a/support/xhpast/Makefile
+++ b/support/xhpast/Makefile
@@ -8,7 +8,7 @@ ifdef DEBUG
 	CPPFLAGS   += -ggdb -DDEBUG
 	FLEXFLAGS  += --debug
 else
-	CPPFLAGS  += -O3 -minline-all-stringops
+	CPPFLAGS   += -O3 $(shell ./bin/xhpast-generate-release-cppflags.php)
 endif
 
 ifdef PROFILE
diff --git a/support/xhpast/bin/xhpast-generate-release-cppflags.php b/support/xhpast/bin/xhpast-generate-release-cppflags.php
new file mode 100755
index 000000000000..be38b249d2cc
--- /dev/null
+++ b/support/xhpast/bin/xhpast-generate-release-cppflags.php
@@ -0,0 +1,15 @@
+#!/usr/bin/env php
+<?php
+
+// $arcanist_root = dirname(dirname(dirname(dirname(__FILE__))));
+// require_once $arcanist_root.'/support/init/init-script.php';
+
+$cppflags = '';
+$arch = php_uname('m');
+
+if ($arch == 'i386' || $arch == 'x86_64') {
+  $cppflags .= ' -minline-all-stringops ';
+}
+
+echo $cppflags;
+echo "\n";
-- 
2.48.1

